src/scheduler/Analysis.py

```diff
@@ -406,19 +418,42 @@ def _get_cumulative_remaining_dependencies(self, scheduled_analyses: Set[str]) -
     # currently running analyses
 
     def _add_to_current_analyses(self, fw_object: Union[Firmware, FileObject]):
-        if isinstance(fw_object, Firmware):
-            self.currently_running[fw_object.uid] = list(fw_object.files_included)
-        elif fw_object.files_included:
-            for parent in self._find_currently_analyzed_parents(fw_object):
-                union = set(fw_object.files_included).union(self.currently_running[parent])
-                self.currently_running[parent] = list(union)
+        try:
+            self.currently_running_lock.acquire()
+            if isinstance(fw_object, Firmware):
+                self.currently_running[fw_object.uid] = {
```

Comment:
then it should start again at 0 but the remaining analyses from the first one should also count towards 100% so it should be counted as complete before all files are processed (which probably doesn't matter since the remaining ones should be in the DB already from the previous analysis)

Reply:
So, in effect. We are correct regarding the completeness of the analysis but still have analysis running that we can see through other information (e.g. unpack queue)?

Reply:
Yes (at least if I'm not mistaken). Feel free to give it a go and see if you can break it.